qa/L0_socket/test.sh

```diff
@@ -0,0 +1,295 @@
+            set +e
+            kill $SERVER_PID
+            wait $SERVER_PID
+            if [ "$?" == "0" ]; then
+                echo -e "\n***\n*** unexpected start $SERVER\n***"
+                cat $SERVER_LOG
+                exit 1
+            fi
+            set -e
+        fi
+
+        # when using different addresses, allow http/grpc port overlap with grpc/http default port
+        if [ "$address" == "explicit" ]; then
+            if [ "$p" == "http" ]; then
+                SERVER_ARGS="--model-repository=$DATADIR $DIFF_EXPLICIT_ADDRESS --http-port 8001"
+            else
+                SERVER_ARGS="--model-repository=$DATADIR $DIFF_EXPLICIT_ADDRESS --grpc-port 8000"
+            fi
+            run_server_nowait
+            sleep 15
+            if [ "$SERVER_PID" == "0" ]; then
+                echo -e "\n***\n*** Failed to start $SERVER\n***"
+                cat $SERVER_LOG
+                exit 1
+            fi
+            kill $SERVER_PID
+            wait $SERVER_PID
+        fi
+
+        # allow http/grpc port overlap with grpc/http explicit if disabled
+        if [ "$p" == "http" ]; then
+            SERVER_ARGS="--model-repository=$DATADIR $SAME_EXPLICIT_ADDRESS --http-port 8007 --grpc-port 8007 --allow-http 0"
+        else
+            SERVER_ARGS="--model-repository=$DATADIR $SAME_EXPLICIT_ADDRESS --grpc-port 8007 --http-port 8007 --allow-grpc 0"
+        fi
+        run_server_nowait
+        sleep 15
+        if [ "$SERVER_PID" == "0" ]; then
+            echo -e "\n***\n*** Failed to start $SERVER\n***"
+            cat $SERVER_LOG
+            exit 1
+        fi
+        kill $SERVER_PID
+        wait $SERVER_PID
+
+        # error if http/grpc port overlaps with grpc/http explicit port 
+        if [ "$p" == "http" ]; then
+            SERVER_ARGS="--model-repository=$DATADIR $SAME_EXPLICIT_ADDRESS --http-port 8003 --grpc-port 8003"
+            run_server_nowait
+            sleep 15
+            if [ "$SERVER_PID" != "0" ]; then
+                set +e
+                kill $SERVER_PID
+                wait $SERVER_PID
+                if [ "$?" == "0" ]; then
+                    echo -e "\n***\n*** unexpected start $SERVER\n***"
+                    cat $SERVER_LOG
+                    exit 1
+                fi
+                set -e
+            fi
+        else
+            # skip, same as http case
+            true
+        fi
+
+        # when using different addresses, allow http/grpc port overlap with grpc/http explicit
+        if [ "$address" == "explicit" ]; then
+            if [ "$p" == "http" ]; then
+                SERVER_ARGS="--model-repository=$DATADIR $DIFF_EXPLICIT_ADDRESS --http-port 8007 --grpc-port 8007"
+            else
+                SERVER_ARGS="--model-repository=$DATADIR $DIFF_EXPLICIT_ADDRESS --grpc-port 8007 --http-port 8007"
+            fi
+            run_server_nowait
+            sleep 15
+            if [ "$SERVER_PID" == "0" ]; then
```

Comment:
Should send ready / live request to make sure the server is running.

Reply:
Updated